### PR TITLE
[Communication] Add variable groups necessary for using new azure subscription

### PR DESCRIPTION
**Pull request by turalf**
We are using the new subscription for our testing. 
The new subscription id is added to the keyvault and this PR changes the tests.yml to access that subscriptionId. 
This PR will be updated in case new variables needed to be able to use the new subscription.

### CODE REVIEW

File: sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     ResponseReplacerProcessor
 )
 
-SKIP_PHONE_NUMBER_TESTS = True
+SKIP_PHONE_NUMBER_TESTS = False
 PHONE_NUMBER_TEST_SKIP_REASON= "Phone Number infra for live tests not ready yet"
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py:19]: nit: remove this comment?

**Review by ankitarorabit — DISMISSED** [review]

**Review by ankitarorabit — APPROVED** [review]

**Comment by turalf** [comment]
azp run python - communication - tests

**Review by ankitarorabit — APPROVED** [review]

**Pull request merged**